### ISSUE TITLE
no-require-imports: False negative when require argument isn't a string literal

### ISSUE THREAD

**Issue by pe8ter** [palantir/tslint#4177]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const filePath = './my-module';
const myModule = require(filePath);
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-require-imports": true
  }
}
```

#### Actual behavior

Linting succeeds.

#### Expected behavior

Linting fails with similar error:

```
ERROR: (no-require-imports) /path/to/your/source.ts[2, 18]: require() style import is forbidden
```

**Comment by pe8ter** [comment]
I tracked the problem down to the _tsutils_ package. While building a list of imports in a source file, this utility decides that an invocation of `require()` is an import and then adds it to a list of imports via a helper function, but this helper function filters out any invocations that do not use a string literal as an argument.

See  https://github.com/ajafff/tsutils/blob/566e35391ed942ec4614dc753b6c0905edf0f7f9/util/util.ts#L1130

I don't know what the reason for this filter is. As long as you invoke `require()` to import a module it shouldn't matter what you pass in. It's still an import.

I will open a bug with that project to hopefully get more input. Let me know if you want me to close this issue, since the root cause is in another repo.

**Comment by ajafff** [comment]
> I will open a bug with that project to hopefully get more input.

I can answer that right here: that function is intended to find all statically known imports, hence the filtering. 
If you need all require calls, that should be raised as a feature request instead of a bug report

**Comment by pe8ter** [comment]
I can make that feature request, but since that change would most likely go into a `3.x` version of _tsutils_, I'd like to know the maintainers of _tslint_ think about making an upgrade from `2.x`.

**Comment by JoshuaKGoldberg** [comment]
Removing the `Type: Breaking Change` label per #4811. Now accepting PRs!

**Comment by adidahiya** [comment]
requires https://github.com/palantir/tslint/issues/4365 first

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
